From b05fe07b919cf666439f262ca05787c41a696e6e Mon Sep 17 00:00:00 2001
From: Egor Pilipenko <pilipenko0409@gmail.com>
Date: Tue, 12 Dec 2023 01:21:48 +0700
Subject: [PATCH] twrp.cpp: Always unmap super devices on fastbootd mode
 proceeding Fix logical partitions flashing in fastbootd mode (was broken in
 662efc0f9bf2ec57dcbb9e7143c939200942ca40) I (1)[557:fastbootd]: Fastboot
 command: flash:system I (1)[557:fastbootd]: [liblp]Updated logical partition
 table at slot 0 on device /dev/block/by-name/super E (1)[557:fastbootd]:
 DM_DEV_CREATE failed for [system]: Device or resource busy E
 (1)[557:fastbootd]: Could not map partition: system

---
 twrp.cpp | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/twrp.cpp b/twrp.cpp
index 5eb59098..cbc2f454 100644
--- a/twrp.cpp
+++ b/twrp.cpp
@@ -115,11 +115,7 @@ static void Decrypt_Page(bool SkipDecryption, bool datamedia) {
 static void process_fastbootd_mode() {
 		LOGINFO("starting fastboot\n");
 
-#ifdef TW_LOAD_VENDOR_MODULES
-		if (android::base::GetBoolProperty("ro.virtual_ab.enabled", false)) {
-			PartitionManager.Unmap_Super_Devices();
-		}
-#endif
+		PartitionManager.Unmap_Super_Devices();
 
 		gui_msg(Msg("fastboot_console_msg=Entered Fastboot mode..."));
 		property_set("ro.orangefox.fastbootd", "1");
-- 
2.40.1
